### ISSUE TITLE
Problem with guest os mappings for Red Hat Enterprise Linux 9 in ACS 4.18.0.0

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#7781]
I checked the databases for RHEL9 guest os mappings and found some issues

1. there are two guest os for Red Hat Enterprise Linux 9
```
mysql> select * from guest_os where display_name like "Red Hat Enterprise Linux 9%";
```
![image](https://github.com/apache/cloudstack/assets/57355700/4f6e9114-132c-447e-819f-ff494a6f26f8)

2. guest os name has extra comma

```
select * from guest_os_hypervisor where guest_os_id=316;
```
![image](https://github.com/apache/cloudstack/assets/57355700/9ce7ff8c-d6bb-4f20-9518-28a231c1ac57)

3. another guest os only applies on VMware 7.0.1.0
```
select * from guest_os_hypervisor where guest_os_id=336;
```
![image](https://github.com/apache/cloudstack/assets/57355700/ac7c7069-7a86-4846-bb47-f17c3a668e1a)

this is not the only one. all guest os mappings in #7095 have the same problem (only supported in VMwre 7.0.1.0).
![image](https://github.com/apache/cloudstack/assets/57355700/0c4cb979-7cda-47f3-8798-47f4e0d908b5)

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
cc @DaanHoogland @shwstppr 

this is related to #7095 and #7059

**Comment by weizhouapache** [comment]
I will address issue 2 in PR #7380

**Comment by DaanHoogland** [comment]
> I will address issue 2 in PR #7380

sorry @weizhouapache, in #7799
- I added a query for this issue 2. We can decide later which to keep.
- I did some work to address the double OSses as well copying mappings from the one to the other, which should solve issue 3.
 I have created a method to deal with issue 1but that still needs to be called in an intelligent way.

cc @BryanMLima

**Comment by weizhouapache** [comment]
fixed by #7799
